lib/semantics/resolve-names.cc

```diff
@@ -3019,6 +3044,51 @@ bool DeclarationVisitor::Pre(const parser::ProcComponentDefStmt &) {
 void DeclarationVisitor::Post(const parser::ProcComponentDefStmt &) {
   interfaceName_ = nullptr;
 }
+bool DeclarationVisitor::Pre(const parser::ProcPointerInit &x) {
+  if (auto *name{std::get_if<parser::Name>(&x.u)}) {
+    return !NameIsKnownOrIntrinsic(*name);
+  }
+  return true;
+}
+bool DeclarationVisitor::Pre(const parser::ProcInterface &x) {
+  if (auto *name{std::get_if<parser::Name>(&x.u)}) {
+    if (!FindSymbol(*name) &&
+        !HandleUnrestrictedSpecificIntrinsicFunction(*name)) {
```

Comment:
Can you use `NameIsKnownOrIntrinsic` here?

Reply:
Yes.  And that leaves `NameIsKnownOrIntrinsic` as the only remaining caller of `HandleUnrestrictedSpecificIntrinsic`, but I think that I'll let it stay as its own function rather than manually folding it into its only caller.